pkg/detectors/deepai/deepai.go

```diff
@@ -61,7 +61,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				continue
 			}
 			writer.Close()
-			req, err := http.NewRequestWithContext(ctx, "POST", "https://api.deepai.org/api/text-tagging", bytes.NewReader(body.Bytes()))
+			req, err := http.NewRequestWithContext(ctx, "POST", "https://api.deepai.org/api/text-tagging", bytes.NewReader([]byte(body.String())))
```

Comment:
What about the fact that we're now converting a string back into bytes? I suspect the benchmarks did not capture the full picture.

Reply:
Yea actually the performance benefit here might not be worth the change.

I was more curious about this portion:
`for _, frag := range file.TextFragments {
			var sb strings.Builder
			newLineNumber := frag.NewPosition
			for _, line := range frag.Lines {
				if line.Op == gitdiff.OpAdd {
					sb.WriteString(line.Line)
				}
			}
			log.WithField("fragment", sb.String()).Trace("detecting fragment")
			metadata := s.sourceMetadataFunc(fileName, email, hash, when, urlMetadata, newLineNumber)
			chunksChan <- &sources.Chunk{
				SourceName:     s.sourceName,
				SourceID:       s.sourceID,
				SourceType:     s.sourceType,
				SourceMetadata: metadata,
				Data:           []byte(sb.String()),
				Verify:         s.verify,
			}
		}`

Because when i was profiling with large repos the WriteString was eating a lot of memory in comparison to strings.Builder.

Reply:
Here even with the additional conversion to []byte(sb.String()) benchmarks show it being significantly faster.

Reply:
![Screen Shot 2022-05-09 at 10 16 59 AM](https://user-images.githubusercontent.com/21311841/167462820-a9cef9de-8301-470f-8835-efedab7342ed.png)